### PR TITLE
script to update dependencies

### PR DESCRIPTION
**Pull request by matthiasdiener**
Not sure if this is useful.

### CODE REVIEW

File: update-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    cd $m
+
+    echo $m
+
+    git pull
+
+    cd ..
+done
+
+echo "Pulled updated modules. Now commit the changes."
```
Comment (inducer) [inline update-dependencies.sh:17]: This is unclear to me. Why would you need to commit after `pull`?

Comment (matthiasdiener) [reply update-dependencies.sh:17]: It's just the way submodules work. Whenever you update a submodule (e.g., through `git pull` or by checking out a specific commit), and you want the change reflected in the parent git repo (emirge in our case), such that a `git clone` of emirge or a `git pull` in emirge also pull the changed submodule versions, you need to commit (and push) the main repo after the modification of the submodules. 

In other words, the submodules in emirge are currently at the version I originally imported them a few weeks ago, since they haven't been updated in emirge until now. This script intends to make this update process a bit easier.

Comment (inducer) [reply update-dependencies.sh:17]: OIC, makes sense.

```suggestion
echo "Pulled updated modules. Now commit the submodule changes to emirge."
```
(I wonder if we should remind them here to create a branch.)

File: update-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -o nounset -o errexit
```
Comment (inducer) [inline update-dependencies.sh:3]: Maybe this should refuse to run if there are uncommitted changes in any sub-repo? (`repo` just skips over repos with changes.)

File: update-dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (inducer) [inline update-dependencies.sh:1]: What do you think of renaming this `pull-submodules.sh`?

File: pull-submodules.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+MY_MODULES=$(git submodule status | awk '{print $2}')
+
+for m in $MY_MODULES; do
+    cd $m
+
+    echo $m
+
+    # Skip directories that have local modifications
+    git diff-index --quiet HEAD || { echo "  Skipping update of '$m' due to local modifications."; cd ..; continue; }
```
Comment (inducer) [inline pull-submodules.sh:13]: `git status --porcelain`

File: pull-submodules.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -o nounset -o errexit
```
Comment (inducer) [inline pull-submodules.sh:3]: Upon second thought, I think this script will be tempting to a lot of students, when the submodule-based route is really what they want. Maybe there should be a short explainer and an "are you sure that this is what you want?" prompt.

File: pull-submodules.sh (hunk 4 above)
Comment (matthiasdiener) [reply pull-submodules.sh:13]: Not sure which one is better. `git diff-index --quiet HEAD` has the advantage (?) that untracked files do not prevent `git pull` from being run.

File: pull-submodules.sh (hunk 5 above)
Comment (matthiasdiener) [reply pull-submodules.sh:3]: Good idea. I added a warning and force the user to run this script with the `-x` option if they are sure they want to execute it.

File: pull-submodules.sh (hunk 4 above)
Comment (inducer) [reply pull-submodules.sh:23]: Either way, as long as you're choosing purposefully.
